controller/backup_controller.go

```diff
@@ -221,6 +270,18 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 		return nil
 	}
 	if !isResponsible {
+		if errIM := bc.ds.CheckDefaultEngineInstanceManagerAvailability(backup.Status.OwnerID); errIM != nil {
+			bc.logger.WithError(errIM).Infof("Transfering backup %v ownership", backup.Name)
+			backup.Status.OwnerID = ""
```

Comment:
Should you just simply make `backup.Status.OwnerID = ""` and update it instead of checking `CheckDefaultEngineInstanceManagerAvailability` again?

You already do the check in `isResponsibleFor`, so I think we can just simply transfer its owner to another node instead.

Reply:
`isReponsibleFor` included other conditions so I am not sure if I should handle them together here. And besides, this will add some log for the transfer.

Reply:
`isReponsibleFor` should determine if the **current** node can/need to take over the ownership. In other words, it's enough to check it in `bc.isResponsibleFor`